frontends/web/src/routes/buy/pocket.tsx

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright 2022 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useTranslation } from 'react-i18next';
+import { useState, useEffect, createRef } from 'react';
+import { MessageVersion, parseMessage, serializeMessage, V0MessageType } from 'request-address';
+import { signAddress, getPocketURL } from '../../api/exchanges';
+import { Header } from '../../components/layout';
+import { Spinner } from '../../components/spinner/Spinner';
+import { useLoad } from '../../hooks/api';
+import { alertUser } from '../../components/alert/Alert';
+import Guide from './guide';
+import style from './iframe.module.css';
+
+interface TProps {
+    code: string;
+}
+
+// TODO:
+// - add T&C
+export const Pocket = ({ code }: TProps) => {
+  const { t } = useTranslation();
+
+  const [height, setHeight] = useState(0);
+  const iframeURL = useLoad(getPocketURL(code));
+
+
+  const ref = createRef<HTMLDivElement>();
+  const iframeRef = createRef<HTMLIFrameElement>();
+  var resizeTimerID: any = undefined;
+
+  const name = 'Bitcoin';
+
+  useEffect(() => {
+    onResize();
+    window.addEventListener('resize', onResize);
+    window.addEventListener('message', onMessage);
+
+    return () => {
+      window.removeEventListener('resize', onResize);
+      window.removeEventListener('message', onMessage);
+    };
+  });
+
+  const onResize = () => {
+    if (resizeTimerID) {
+      clearTimeout(resizeTimerID);
+    }
+    resizeTimerID = setTimeout(() => {
+      if (!ref.current) {
+        return;
+      }
+      setHeight(ref.current.offsetHeight);
+    }, 200);
+  };
+
+  const onMessage = (e: MessageEvent) => {
+    if (!iframeURL) {
+      return;
+    }
+
+    const iframeOrigin = iframeURL.match(/^https?:\/\/[^/?]+/)![0];
```

Comment:
Maybe change to `new URL(location.href).origin`. If the url can be invalid it should be in a try/catch, but I think in this case it is fine.

Reply:
Ah nice! Thank you :) Now should be better